### PR TITLE
feat: implement globalThis.navigator.userAgent

### PR DESCRIPTION
**Pull request by nabetti1720**
### Description of changes

[The Minimum Common Web Platform API](https://common-min-api.proposal.wintercg.org/) is being considered at WinterCG.
I implemented one of the APIs, `globalThis.navigator.userAgent`.

This is my first time coding in the rust language, so I will promote it as a draft version.
Please point out if it does not follow `llrt` or `rust` conventions.

### Local execution results

image

### Checklist

- [x] Created unit tests in `tests/unit` and/or in Rust for my feature if needed
- [x] Ran `make fix` to format JS and apply Clippy auto fixes
- [x] Made sure my code didn't add any additional warnings: `make check`
- [x] Updated documentation if needed ([API.md](API.md)/[README.md](README.md)/Other)

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: src/http/fetch.rs [hunk 1]
```diff
@@ -160,7 +160,7 @@ pub(crate) fn init(ctx: &Ctx<'_>, globals: &Object) -> Result<()> {
                 let mut req = Request::builder()
                     .method(method)
                     .uri(uri)
-                    .header("user-agent", format!("llrt {}", VERSION))
+                    .header("user-agent", format!("llrt/{}", VERSION))
```
Comment (richarddavison) [inline src/http/fetch.rs:163]: nice finding!

File: tests/unit/navigator.test.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import defaultImport from "navigator";
+import * as namedImport from "navigator";
+
+describe("navigator.userAgent", () => {
+  it("should have a navigator userAgent", () => {
+    expect(defaultImport.userAgent).toEqual(navigator.userAgent);
```
Comment (richarddavison) [inline tests/unit/navigator.test.ts:6]: should we maybe do a regex check here or starts with "llrt/"?

**Review by richarddavison — CHANGES_REQUESTED** [review]
Looks great! Minor test comment

File: tests/unit/navigator.test.ts (hunk 2 above)
Comment (nabetti1720) [reply tests/unit/navigator.test.ts:6]: Thanks for pointing that out.
According to the specification, it is not required to start with a slash, but we will follow the implementation and make sure that it starts with "llrt/".

Comment (richarddavison) [reply tests/unit/navigator.test.ts:6]: Ok if that's not the case they we should revert to `llrt {version}` because this is already used for stats :)

Comment (nabetti1720) [reply tests/unit/navigator.test.ts:6]: OK, I understand.
I'll try to revert to `llrt {version}`.

**Review by richarddavison — APPROVED** [review]

**Pull request merged**